notificationservice/src/main/java/com/example/notificationservice/services/blacklist/BlacklistService.java

```diff
@@ -0,0 +1,56 @@
+package com.example.notificationservice.services.blacklist;
+
+import com.example.notificationservice.models.Blacklist.BlacklistRecord;
+import com.example.notificationservice.models.responsebody.BlacklistResponseBody;
+import com.example.notificationservice.repositories.BlacklistRepository;
+import com.example.notificationservice.repositories.RedisCacheManager;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class BlacklistService {
+    private final BlacklistRepository blacklistRepository;
+    private final RedisCacheManager redisCacheManager;
+    private  final String REDIS_CACHE_NAME = "blacklist_numbers";
+
+    public BlacklistResponseBody getBlacklist() {
+        return BlacklistResponseBody.builder().phoneNumbers(redisCacheManager.findAll(REDIS_CACHE_NAME)).build();
+    }
+
+
+    public BlacklistResponseBody addToBlacklist(List<String> phoneNumbers) throws Exception{
+        if ((phoneNumbers != null) && (phoneNumbers.isEmpty())) {
```

Comment:
use this in controller
if(CollectionUtils.isEmpty(requestBody.getPhoneNumbers())){
                throw new InvalidRequestException("No number to add");
            }